### PR TITLE
Add Ephemeral::{metadata, targets}

### PR DESCRIPTION
**Pull request by erickt**
This changes Ephemeral repository to allow access to the underlying metadata
and targets that were written to the repository. This allows for direct
modification, in case users want to directly add metadata to the repository.
This should be safe, because the repositories are considered untrusted, and
still needs to go through verification through the workflow before it is used
by the client.

### CODE REVIEW

File: src/repository/ephemeral.rs [hunk 1]
```diff
@@ -37,6 +37,16 @@ where
             _interchange: PhantomData,
         }
     }
+
+    /// Return a map for all the metadata in the repository.
+    pub fn metadata(&self) -> ArcHashMap<(MetadataPath, MetadataVersion), Arc<[u8]>> {
```
Comment (wellsie1116) [inline src/repository/ephemeral.rs:42]: Will rustdoc make one of those unclickable links if ArcHashMap is not pub?  If so, consider exporting that type alias or expanding the alias here.

Comment (erickt) [reply src/repository/ephemeral.rs:42]: I just checked, and rustdoc will automatically expand these type aliases to `Arc<RwLock<HashMap<...>>>`.

**Review by wellsie1116 — APPROVED** [review]
Agreed this seems safe, it'd be possible to do this kind of thing already with a tempdir and FileSystemRepository.

I think a builder and/or more scoped CRUD APIs would be a bit nicer since it wouldn't need to expose data structure details of EphemeralRepository in the public interface, but it probably isn't worth the extra plumbing.

**Comment by erickt** [comment]
Ended up doing this in a separate CL.

**Pull request closed without merging**